deepchecks/vision/utils/test_utils.py

```diff
@@ -9,48 +9,118 @@
+    indices = []
+    batch_sampler = data_loader.batch_sampler
+    # Using the batch sampler to get all indices
+    for batch in batch_sampler:
+        indices += batch
+    if shuffle:
+        indices = random.sample(indices, len(indices))
+    if n_samples is not None:
+        indices = indices[:n_samples]
+
+    # Create new sampler and batch sampler
+    sampler = IndicesSequentialSampler(indices)
+    new_batch_sampler = BatchSampler(sampler, batch_sampler.batch_size, batch_sampler.drop_last)
+
+    props = _get_data_loader_props(data_loader)
+    props['batch_sampler'] = new_batch_sampler
+    _collisions_removal_dataloader_props(props)
+    return data_loader.__class__(**props)
+
+
+class IndicesSequentialSampler(Sampler):
+    """Samples elements sequentially from a given list of indices, without replacement.
+    Args:
+        indices (sequence): a sequence of indices
     """
 
-    class ModifiedDataset():
-        """A modified dataset object, returning func_to_apply for each index."""
+    indices: List[int]
+
+    def __init__(self, indices: List[int]) -> None:
+        super().__init__(None)
+        self.indices = indices
+
+    def __iter__(self) -> Iterator[int]:
+        """Return an iterator over the indices."""
+        return iter(self.indices)
 
-        def __init__(self, orig_dataset):
-            self._orig_dataset = orig_dataset
+    def __len__(self) -> int:
+        """Return the number of indices."""
+        return len(self.indices)
 
-        def __getitem__(self, idx):
-            return func_to_apply(self._orig_dataset, idx)
+    def index_at(self, location):
+        """Return for a given location, the real index value."""
+        return self.indices[location]
 
-        def __len__(self):
-            return len(self._orig_dataset)
 
-    # Code needed to return a dataloader with the modified dataset that is otherwise identical to the original.
-    props = vision_data._get_data_loader_props(vision_data.data_loader)  # pylint: disable=protected-access
-    props['dataset'] = ModifiedDataset(vision_data.data_loader.dataset)
-    props['shuffle'] = shuffle
-    data_loader = DataLoader(**props)
-    data_loader, _ = vision_data._get_data_loader_sequential(data_loader)  # pylint: disable=protected-access
-    return data_loader
+def un_normalize_batch(tensor: torch.Tensor, mean: Sized, std: Sized, max_pixel_value: int = 255):
+    """Apply un-normalization on a tensor in order to display an image."""
+    dim = len(mean)
+    reshape_shape = (1, 1, 1, dim)
+    max_pixel_value = [max_pixel_value] * dim
+    mean = torch.tensor(mean, device=tensor.device).reshape(reshape_shape)
+    std = torch.tensor(std, device=tensor.device).reshape(reshape_shape)
+    tensor = (tensor * std) + mean
+    tensor = tensor * torch.tensor(max_pixel_value, device=tensor.device).reshape(reshape_shape)
+    return tensor.cpu().detach().numpy()
```

Comment:
can you fix all tensor.cpu().detach().numpy() which aren't like that?
like fix all tensor.cpu().numpy() / tensor.numpy() / tensor.detach().cpu().numpy()
(not related to here but made me remember)